larky-api/src/main/java/com/verygood/security/larkyapi/VersionedLarkyEngineImpl.java

```diff
@@ -0,0 +1,224 @@
+package com.verygood.security.larkyapi;
+
+import org.springframework.core.io.Resource;
+import org.springframework.core.io.support.PathMatchingResourcePatternResolver;
+
+import javax.annotation.Nonnull;
+import javax.script.CompiledScript;
+import javax.script.ScriptContext;
+import javax.script.ScriptEngine;
+import javax.script.ScriptException;
+import javax.script.Bindings;
+import javax.script.ScriptEngineFactory;
+import java.io.Reader;
+import java.lang.reflect.InvocationTargetException;
+import java.lang.reflect.Method;
+import java.net.MalformedURLException;
+import java.net.URISyntaxException;
+import java.net.URL;
+import java.net.URLClassLoader;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Set;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+public class VersionedLarkyEngineImpl implements VersionedLarkyEngine {
+
+  final private Class engineClass;
+  final private Class scriptClass;
+  final private Class parseClass;
+  final private ScriptEngine engineInstanceObj;
+  final private String version;
+
+
+  // Instantiating the static map
+  private static Map<String, URL> larkyJarByVersion;
+  private static void detectVersions() {
+    larkyJarByVersion = new HashMap<>();
+    try {
+      PathMatchingResourcePatternResolver resolver =
+              new PathMatchingResourcePatternResolver(VersionedLarkyEngineImpl.class.getClassLoader());
+      // Get jars with format `larky-\d{1,3}.\d{1,3}.\d{1,3}-fat.jar`
+      Resource[] resources = resolver.getResources( // uses AntPathMatcher
+              "classpath*:larky-{\\d+}.{\\d+}.{\\d+}-fat.jar"
```

Comment:
Update the definition for the agreed format

Reply:
done